### PR TITLE
Generate systemd.directive entries from dbus documentation

### PR DESCRIPTION
**Pull request by boucman**
This PR modifies the tooling to automatically generate systemd.directives entries for
* bus methods
* bus signals
* bus properties

the first patch is, strictly speaking, not needed, but the feature is trivial, can be usefull and was well tested in my previous attempt at fixing this problem

I did not run the update-dbus-docs in this PR because it needs a specific setup (session number, containers) that I don't have an my results were different than what is needed. I'll leave it to @keszybz to refresh that...

### CODE REVIEW

**Comment by keszybz** [comment]
At the XML level, I get the following diff:
```diff
@@ -71,22 +71,26 @@ node /org/freedesktop/timedate1 {
   interface org.freedesktop.DBus.Properties { ... };
 };
     </programlisting>
-
-    <!--method ListTimezones is not documented!-->
-
-    <!--property Timezone is not documented!-->
-
-    <!--property LocalRTC is not documented!-->
-
-    <!--property CanNTP is not documented!-->
-
-    <!--property NTP is not documented!-->
-
-    <!--property NTPSynchronized is not documented!-->
-
-    <!--property TimeUSec is not documented!-->
-
-    <!--property RTCTimeUSec is not documented!-->
+<!--method ListTimezones is not documented!-->
+<!--property Timezone is not documented!-->
+<!--property LocalRTC is not documented!-->
+<!--property CanNTP is not documented!-->
+<!--property NTP is not documented!-->
+<!--property NTPSynchronized is not documented!-->
+<!--property TimeUSec is not documented!-->
+<!--property RTCTimeUSec is not documented!-->
+<variablelist class="dbus-method" generated="True" extra-ref="SetTime"/>
+<variablelist class="dbus-method" generated="True" extra-ref="SetTimezone"/>
+<variablelist class="dbus-method" generated="True" extra-ref="SetLocalRTC"/>
+<variablelist class="dbus-method" generated="True" extra-ref="SetNTP"/>
+<variablelist class="dbus-method" generated="True" extra-ref="ListTimezones"/>
+<variablelist class="dbus-property" generated="True" extra-ref="Timezone"/>
+<variablelist class="dbus-property" generated="True" extra-ref="LocalRTC"/>
+<variablelist class="dbus-property" generated="True" extra-ref="CanNTP"/>
+<variablelist class="dbus-property" generated="True" extra-ref="NTP"/>
+<variablelist class="dbus-property" generated="True" extra-ref="NTPSynchronized"/>
+<variablelist class="dbus-property" generated="True" extra-ref="TimeUSec"/>
+<variablelist class="dbus-property" generated="True" extra-ref="RTCTimeUSec"/>
 
     <refsect2>
       <title>Methods</title>
```
The change in *vertical* whitespace is OK. But please fix the lack of indentation.

Maybe insert a comment that the following list of `<variablelist>`s is there to provide a hook for indexing, and does not result in any visible output? This might not be obvious to someone who tries to edit the man page.

On the formatted level, I get the following difference:
```diff
+DBUS METHODS
+       Methods exposed on dbus.
+
+       ListTimezones
+           org.freedesktop.timedate1(5)
+
+       SetLocalRTC
+           org.freedesktop.timedate1(5)
+
+       SetNTP
+           org.freedesktop.timedate1(5)
+
+       SetTime
+           org.freedesktop.timedate1(5)
+
+       SetTimezone
+           org.freedesktop.timedate1(5)
...
```
1. In the title: dbus → D-Bus
2. "Methods exposed on dbus." → "Methods exposed in the D-Bus interface."
3. I think it'd be more useful to have the full method name, i.e. `org.freedesktop.timedate1.SetLocalRTC`. I'm open to counterarguments on this one ;)
4. Method names should be suffixed by ().

**Comment by boucman** [comment]
I'll see what I can do... most of the remarks shouldn't be too complicated..

I don't understand how I broke the indentation of the comments, but i'll check that.

**Comment by boucman** [comment]
For your latest remark, the method name is SetLocalRtc... org.freedesktop.timedate1.SetRTC is the interface name followed by the method name...

More seriously, though...I think adding the interface name makes the resulting systemd.directives less readable, and its not that usefull in a sense (when you know the interface name, you know the man page to look for)

I've added a new section cross-referencing interfaces, so that shoulde cover it I think....

Everything else was fixed as you requested, PTAL

File: tools/make-directive-index.py [hunk 1]
```diff
@@ -160,6 +160,38 @@
                 <variablelist id='filenames' />
         </refsect1>
 
+        <refsect1>
+                <title>d-bus interfaces</title>
```
Comment (keszybz) [inline tools/make-directive-index.py]: D-bus! (capital "D")

File: tools/make-directive-index.py [hunk 2]
```diff
@@ -160,6 +160,38 @@
...
+
+                <para>Interaces exposed on the D-BUS.</para>
```
Comment (keszybz) [inline tools/make-directive-index.py]: "Interfaces exposed over D-Bus."
(D-Bus is a name, hence "the" should not be used.)

File: tools/make-directive-index.py [hunk 3]
```diff
@@ -160,6 +160,38 @@
...
+
+                <variablelist id='dbus-interface' />
+        </refsect1>
+
+        <refsect1>
+                <title>d-bus methods</title>
+
+                <para>Methods exposed in the D-BUS interface.</para>
```
Comment (keszybz) [inline tools/make-directive-index.py]: "D-Bus"

**Comment by keszybz** [comment]
Seems to work great. I think this is good for merging once the spelling issues are resolved ;)

**Comment by keszybz** [comment]
Maybe also run `tools/update-dbus-docs.py` over the man pages? We need to do that sooner or later.

**Comment by boucman** [comment]
Done... I did not run update-dbus-docs because my current setup gives incorrect results on master.

Sorry about that, hope this is not too much of an issue...

**Comment by keszybz** [comment]
> Done... I did not run update-dbus-docs because my current setup gives incorrect results on master.
> 
> Sorry about that, hope this is not too much of an issue...

Np., I can do that here.

**Pull request merged**